### ISSUE TITLE
MIT Living Wage Costs Increase

### ISSUE THREAD

**Issue by vaibhavb007** [CSStipendRankings/CSStipendRankings#98]
Hello
I am a Georgia Tech student and noticed that last month MIT living wage increased the cost for Fulton County (where GT is located) from $39k to $54.5k. I am assuming there would be similar increases for other universities/counties and I wanted to let you know about the same.

**Comment by mjc0608** [comment]
Yeah currently the living wage is updated manually. I don't know whether we should come up with a more scalable way to do this.

**Comment by naman** [comment]
For the time being, as incoming students are making decisions, I think it would be helpful to add a comment on the website along the lines ...
```
Living cost is calculated based on the [MIT Living Wage Calculator](https://livingwage.mit.edu/)for the institution's city. The calculations were last pulled on Month, Year and we recognize that the living wages keep changing according to current world economic conditions. See why we use this calculator and its limitations in [here](https://csstipendrankings.org/faq.html#:~:text=Why%20MIT%20Living%20Cost%20Calculator?).
```

**Comment by mjc0608** [comment]
@naman I submitted the request form I found in their website. However, it's not the same as the one you provided. Also, I added you to the CSStipendRankings org and you should have write access to this repo now.

**Comment by mjc0608** [comment]
Also @naman, we do not pull the living wage but relies on submitters to do that work. So there's not such a single time that the data was pulled.

**Comment by naman** [comment]
Fantastic! Thanks for filling the form. In case we get the data for all the counties, we can pull the updated living wage data quarterly/yearly automatically and perhaps rely on the submitters to submit PR for the stipends?

**Comment by naman** [comment]
@vaibhavb007 in the meanwhile, can you submit a PR to update the GT living wage? I am happy to verify the data.

**Comment by vaibhavb007** [comment]
I completed that. However, let me also add that having just Georgia Tech's cost updated paints Georgia Tech in a bad light. We should work on doing this for other universities as well, even if it is done manually for this one time.
I can definitely volunteer a few hours to do this.

**Comment by naman** [comment]
@vaibhavb007 thanks, please feel free to make a PR to update other universities too

**Comment by JennLCSD** [comment]
This has been sitting since April and the data update for CoL remains random. The list is misleading.

I'm planning to do a pull request to update all the seriously out-of-date cost of living amounts. (I've checked the entire list against the MIT calculator)

- Despite the caveat "This ranking is part informational and part satire, designed to democratize information on how PhD students in computer science and related areas are paid for their labor..." this tool is being referred to in negotiation and for direct comparison between institutions and would better serve the intended users if it were consistently updated with the known "Cost of Living" data set specifically. 
- Only 6 institutions living expenses have been updated to match current MIT calculator info.
- Presumably stipend increases have been applied without updating the cost of living:
  - If CoL is not automated shouldn't these be linked? 
  - If a stipend is updated CoL should be confirmed to have been updated to most current MIT info?
  - Why should we trust the stipend/living cost relationship overall if stipend can be updated without confirming the CoL?
- With awareness that this is volunteer maintained:
  - What academic year/term/semester does a given line or column of data represent? Could this be added/indicated somewhere?
  - How does a general viewer seeking comparison info know *when* data are updated? Could "updated" date stamps be transparently added?
- Example issue as of June 20, 2024: if Yale's stipend is in fact $48,300 and the cost of living is updated to currently available MIT data ($48,247), then the difference is only $83 *not* $11K. If someone puts in a request for Yale's *stipend* to be updated with an increase but the *cost of living* is not correspondingly confirmed/updated this inaccuracy would grow.
- Listing fellowships confuses the issue — example: NSF GRP at $37,000. The cost of living entered is for Pittsburgh. GRP, Hertz are not standard base stipends provided by a university and can be granted to students at a various of institutions with a wide range in CoL.

P.S. University of Pittsburgh ... ends with an "h". ;-)

**Comment by mjc0608** [comment]
> I'm planning to do a pull request to update all the seriously out-of-date cost of living amounts.
Thanks for the commit. I will add you as a project owner once the pull request is merged.

> If CoL is not automated shouldn't these be linked
MIT living wage calculator does not have an API. We rely on students from the institution to provide such data. I don't think it's legal to fetch data from MIT living wage calculator without their agreement. They didn't reply our email requesting for data.

> What academic year/term/semester does a given line or column of data represent? Could this be added/indicated somewhere?
We will need the data, and there will be backward compatibility issues. We assume people will always update the list to make it up-to-date. Our assumption is if the data in our website is important enough for negotiation purpose, people will update it.

> How does a general viewer seeking comparison info know when data are updated? Could "updated" date stamps be transparently added?
This information is in the commit history. Personally I think we already have enough numbers in the table and I do not want to add more.

> Listing fellowships confuses the issue — example: NSF GRP at $37,000...
You can set the university when you enable fellowship.

**Comment by mjc0608** [comment]
> If you want to update a stipend you should be encouraged to confirm the cost of living has been updated.

Yeah I'm thinking about a better way for data update. E.g., a javascript form that you can write down the information and it will generate the data entry for you. So people can create pull requests instead of issues, which makes the maintenance  easier. I don't have an estimation when this would be implemented though.

For now, maybe we can update the issue template.

> The number of pulls and updates don't give me confidence regarding this expectation, but I'm happy to pay attention to cost of living. Perhaps it will spur institutions to update stipend info if there is glaring inaccuracy on that side of the equation.

This is a pretty significant commitment. Thanks for this.

For the long run, maybe it would be better to have data regarding the university's zipcode, so if we get data source in the future we can integrate it.

> I was hoping for a single "CoL updated June 2024" based on last merged/approved - or something along those lines.

We can probably add a column for last updated date, and show an icon based on whether it's recently updated in javascript. Hopefully this can be implemented together with the data entry generating script.

**Comment by manathan1984** [comment]
Great discussion, everyone. I have a suggestion. I was wondering if https://www.epi.org/resources/budget/ has a programmatic API that the website may be able to pull periodically. Alternatively, they have all their data available: [full dataset](https://files.epi.org/uploads/fbc_data_2024.xlsx) and [living wage standards](https://files.epi.org/uploads/fbc_livingwage_data_2024.xlsx). One could either use the 1 adult 0 children column uniformly (despite this being higher than the MIT living wage, to ensure fair comparison). Further, it may be meaningful to have a either a weight or edit the raw data to adjust things that maybe overestimated for the case of grad students (maybe transportation). I think the point is to make sure the comparisons of salaries across schools are meaningful.

**Comment by mjc0608** [comment]
Thanks @manathan1984, this is indeed a great data source.

**Comment by mjc0608** [comment]
@claude Take a look into this issue, especially for the last few messages. Especially,

1. Take a look into https://www.epi.org/resources/budget/  and find the download link of the family budget calculator full dataset. Download the dataset and analyze how we can use it. You should focus on the 1p0c data.
2. Edit the website, add a new column (or a separate table) to place the necessary new data. In the frontend, the website can grab the data and let user choose between two data sources (i.e., MIT living wage and this new one). Do not override existing data.
3. Evaluate the solution and open a PR for it.

Be sure to make plan and execute
